### PR TITLE
Re-write cisco_command_config using new beaker template

### PR DESCRIPTION
**Pull request by mikewiebe**
This is a complete re-write of the `cisco_command_config` beaker provider tests. 

1) New test is easier to extend for both content and platform coverage.
2) We now have idempotence testing.
3) Test uses `cisco_command_config` to configure the device and then uses existing providers `cisco_bgp` and `cisco_interface` to verify the config applied.

**N9K Beaker Tests**
```
Beaker::Hypervisor, found some none boxes to create
No tests to run for suite 'pre_suite'
Begin cisco_command_config/test_command_config.rb

TestCase :: Resource cisco_command_config

------------------------------------------------------------
Apply cisco_command_config test.

  * TestStep :: absent clean
  * absent clean Removing cisco_bgp '55 default'

  * TestStep :: absent clean
  * absent clean Removing cisco_interface 'vlan1'

  * TestStep :: 1.1 Apply BGP Config :: MANIFEST    
1.1 Apply BGP Config :: MANIFEST     :: PASS

  * TestStep :: 1.1 Apply BGP Config :: RESOURCE    
1.1 Apply BGP Config :: RESOURCE     :: PASS

  * TestStep :: 1.1 Apply BGP Config :: IDEMPOTENCE 
1.1 Apply BGP Config :: IDEMPOTENCE  :: PASS

  * TestStep :: 1.2 Apply INTERFACE Config :: MANIFEST    
1.2 Apply INTERFACE Config :: MANIFEST     :: PASS

  * TestStep :: 1.2 Apply INTERFACE Config :: RESOURCE    
1.2 Apply INTERFACE Config :: RESOURCE     :: PASS

  * TestStep :: 1.2 Apply INTERFACE Config :: IDEMPOTENCE 
1.2 Apply INTERFACE Config :: IDEMPOTENCE  :: PASS
TestCase :: # {testheader} :: End
cisco_command_config/test_command_config.rb passed in 55.17 seconds
      Test Suite: tests @ 2016-01-23 11:00:11 -0500

      - Host Configuration Summary -


              - Test Case Summary for suite 'tests' -
       Total Suite Time: 55.17 seconds
      Average Test Time: 55.17 seconds
              Attempted: 1
                 Passed: 1
                 Failed: 0
                Errored: 0
                Skipped: 0
                Pending: 0
                  Total: 1

      - Specific Test Case Status -
```

**N6K Beaker Tests**
```
Beaker::Hypervisor, found some none boxes to create
No tests to run for suite 'pre_suite'
Begin cisco_command_config/test_command_config.rb

TestCase :: Resource cisco_command_config

------------------------------------------------------------
Apply cisco_command_config test.

  * TestStep :: absent clean
  * absent clean Removing cisco_bgp '55 default'

  * TestStep :: absent clean
  * absent clean Removing cisco_interface 'loopback1'

  * TestStep :: 1.1 Apply BGP Config :: MANIFEST    
1.1 Apply BGP Config :: MANIFEST     :: PASS

  * TestStep :: 1.1 Apply BGP Config :: RESOURCE    
1.1 Apply BGP Config :: RESOURCE     :: PASS

  * TestStep :: 1.1 Apply BGP Config :: IDEMPOTENCE 
1.1 Apply BGP Config :: IDEMPOTENCE  :: PASS

  * TestStep :: 1.2 Apply INTERFACE Config :: MANIFEST    
1.2 Apply INTERFACE Config :: MANIFEST     :: PASS

  * TestStep :: 1.2 Apply INTERFACE Config :: RESOURCE    
1.2 Apply INTERFACE Config :: RESOURCE     :: PASS

  * TestStep :: 1.2 Apply INTERFACE Config :: IDEMPOTENCE 
1.2 Apply INTERFACE Config :: IDEMPOTENCE  :: PASS
TestCase :: # {testheader} :: End
cisco_command_config/test_command_config.rb passed in 99.57 seconds
      Test Suite: tests @ 2016-01-23 11:04:10 -0500

      - Host Configuration Summary -


              - Test Case Summary for suite 'tests' -
       Total Suite Time: 99.57 seconds
      Average Test Time: 99.57 seconds
              Attempted: 1
                 Passed: 1
                 Failed: 0
                Errored: 0
                Skipped: 0
                Pending: 0
                  Total: 1

      - Specific Test Case Status -
```

**N7K Beaker Tests**
```
Beaker::Hypervisor, found some none boxes to create
No tests to run for suite 'pre_suite'
Begin cisco_command_config/test_command_config.rb

TestCase :: Resource cisco_command_config

------------------------------------------------------------
Apply cisco_command_config test.

  * TestStep :: absent clean
  * absent clean Removing cisco_bgp '55 default'

  * TestStep :: absent clean
  * absent clean Removing cisco_interface 'port-channel10'
  * absent clean Removing cisco_interface 'port-channel100'

  * TestStep :: 1.1 Apply BGP Config :: MANIFEST    
1.1 Apply BGP Config :: MANIFEST     :: PASS

  * TestStep :: 1.1 Apply BGP Config :: RESOURCE    
1.1 Apply BGP Config :: RESOURCE     :: PASS

  * TestStep :: 1.1 Apply BGP Config :: IDEMPOTENCE 
1.1 Apply BGP Config :: IDEMPOTENCE  :: PASS

  * TestStep :: 1.2 Apply INTERFACE Config :: MANIFEST    
1.2 Apply INTERFACE Config :: MANIFEST     :: PASS

  * TestStep :: 1.2 Apply INTERFACE Config :: RESOURCE    
1.2 Apply INTERFACE Config :: RESOURCE     :: PASS

  * TestStep :: 1.2 Apply INTERFACE Config :: IDEMPOTENCE 
1.2 Apply INTERFACE Config :: IDEMPOTENCE  :: PASS
TestCase :: # {testheader} :: End
cisco_command_config/test_command_config.rb passed in 77.30 seconds
      Test Suite: tests @ 2016-01-23 11:07:30 -0500

      - Host Configuration Summary -


              - Test Case Summary for suite 'tests' -
       Total Suite Time: 77.30 seconds
      Average Test Time: 77.30 seconds
              Attempted: 1
                 Passed: 1
                 Failed: 0
                Errored: 0
                Skipped: 0
                Pending: 0
                  Total: 1

      - Specific Test Case Status -

```

### CODE REVIEW

File: tests/beaker_tests/cisco_command_config/test_command_config.rb [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+#
+tests['configure_bgp'] = {
+  puppet_resource: 'cisco_bgp',
+  platform:        'n(3|5|6|7|9)k',
+  # Command indentation is very important!
+  # Make sure config appears exactly how it nvgens on the switch.
+  manifest_props:  "
+    command                  => '
+      feature bgp
+      router bgp 55
```
Comment (glennmatthews) [inline tests/beaker_tests/cisco_command_config/test_command_config.rb]: Might be nice to include some additional submodes (af, neighbor, neighbor_af) in order to more thoroughly validate that command-config handles indentation correctly?

File: tests/beaker_tests/cisco_command_config/test_command_config.rb [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+#   that permits re-use of the properties for both :present and :absent testing
+#   without destroying the hash
+# tests[id][:puppet_resource] - Puppet resource used to verify the configuration
+#   applied by the cisco_command_config resource.
+#
+tests['configure_bgp'] = {
+  puppet_resource: 'cisco_bgp',
+  platform:        'n(3|5|6|7|9)k',
+  # Command indentation is very important!
+  # Make sure config appears exactly how it nvgens on the switch.
```
Comment (glennmatthews) [inline tests/beaker_tests/cisco_command_config/test_command_config.rb]: `s/switch/node/g` - XR nodes are routers, not switches. :-)

Comment (mikewiebe) [inline tests/beaker_tests/cisco_command_config/test_command_config.rb]: Good call.

**Comment by chrisvanheuveln** [comment]
sorry - I added comments on the diff link instead of the files changed tab. sigh...

File: tests/beaker_tests/cisco_command_config/test_command_config.rb (hunk 1 above)
Comment (mikewiebe) [inline tests/beaker_tests/cisco_command_config/test_command_config.rb]: Good idea.

**Comment by chrisvanheuveln** [comment]
:+1:

**Comment by mikewiebe** [comment]
@chrisvanheuveln @glennmatthews Review comments addressed.

**Comment by glennmatthews** [comment]
:+1:

**Pull request merged**